<?xml version="1.0"?>
<launch>
    <arg name="MODE" default="0" />
    <arg name="ROBOTS_NUM" default="20" />
    <arg name="ROBOT_FOV" default="360.0" />
    <arg name="ROBOT_RANGE" default="15.0" />
    <arg name="AREA_SIZE_x" default="60.0" />
    <arg name="AREA_SIZE_y" default="60.0" />
    <arg name="AREA_LEFT" default="-30.0" />
    <arg name="AREA_BOTTOM" default="-30.0" />
    <arg name="GRAPHICS_ON" default="false" />
    <arg name="save_logs" default="true" />
    <arg name="save_cpu" default="false" />
    <arg name="GAUSS_X" default="-8.0" />
    <arg name="GAUSS_Y" default="-8.0" />

    <arg name="use_robot_0" default="1" />
    <arg name="use_robot_1" default="1" />
    <arg name="use_robot_2" default="1" />
    <arg name="use_robot_3" default="1" />
    <arg name="use_robot_4" default="1" />
    <arg name="use_robot_5" default="1" />
    <arg name="use_robot_6" default="0" />
    <arg name="use_robot_7" default="0" />
    <arg name="use_robot_8" default="0" />
    <arg name="use_robot_9" default="0" />



    <node name="centr_coverage"  pkg="coverage_unimore_nyu" type="coverage_centralized" output="screen">
        <param name="ROBOT_ID" value="0" />
        <param name="MODE"  value="$(arg MODE)" />
        <param name="ROBOTS_NUM"  value="$(arg ROBOTS_NUM)" />
        <param name="ROBOT_FOV"  value="$(arg ROBOT_FOV)" />
        <param name="ROBOT_RANGE"  value="$(arg ROBOT_RANGE)" />
        <param name="AREA_SIZE_x"  value="$(arg AREA_SIZE_x)" />
        <param name="AREA_SIZE_y"  value="$(arg AREA_SIZE_y)" />
        <param name="AREA_LEFT"  value="$(arg AREA_LEFT)" />
        <param name="AREA_BOTTOM"  value="$(arg AREA_BOTTOM)" />
        <param name="GRAPHICS_ON"  value="true" />
        <param name="SAVE_POS"  value="true" />
        <param name="SAVE_CPU_TIME"  value="$(arg save_cpu)" />
        <param name="GAUSS_X" value="$(arg GAUSS_X)" />
        <param name="GAUSS_Y" value="$(arg GAUSS_Y)" />

        <remap from="/turtlebot0/odom" to="/hummingbird0/ground_truth/odometry"/>
        <remap from="/turtlebot1/odom" to="/hummingbird1/ground_truth/odometry"/>
        <remap from="/turtlebot2/odom" to="/hummingbird2/ground_truth/odometry"/>
        <remap from="/turtlebot3/odom" to="/hummingbird3/ground_truth/odometry"/>
        <remap from="/turtlebot4/odom" to="/hummingbird4/ground_truth/odometry"/>
        <remap from="/turtlebot5/odom" to="/hummingbird5/ground_truth/odometry"/>
        <remap from="/turtlebot6/odom" to="/hummingbird6/ground_truth/odometry"/>
        <remap from="/turtlebot7/odom" to="/hummingbird7/ground_truth/odometry"/>
        <remap from="/turtlebot8/odom" to="/hummingbird8/ground_truth/odometry"/>
        <remap from="/turtlebot9/odom" to="/hummingbird9/ground_truth/odometry"/>
        <remap from="/turtlebot10/odom" to="/hummingbird10/ground_truth/odometry"/>
        <remap from="/turtlebot11/odom" to="/hummingbird11/ground_truth/odometry"/>
        <remap from="/turtlebot12/odom" to="/hummingbird12/ground_truth/odometry"/>
        <remap from="/turtlebot13/odom" to="/hummingbird13/ground_truth/odometry"/>
        <remap from="/turtlebot14/odom" to="/hummingbird14/ground_truth/odometry"/>
        <remap from="/turtlebot15/odom" to="/hummingbird15/ground_truth/odometry"/>
        <remap from="/turtlebot16/odom" to="/hummingbird16/ground_truth/odometry"/>
        <remap from="/turtlebot17/odom" to="/hummingbird17/ground_truth/odometry"/>
        <remap from="/turtlebot18/odom" to="/hummingbird18/ground_truth/odometry"/>
        <remap from="/turtlebot19/odom" to="/hummingbird19/ground_truth/odometry"/>
        <remap from="/turtlebot20/odom" to="/hummingbird20/ground_truth/odometry"/>
        


        <remap from="/turtlebot0/cmd_vel" to="/hummingbird0/autopilot/velocity_command"/>
        <remap from="/turtlebot1/cmd_vel" to="/hummingbird1/autopilot/velocity_command"/>
        <remap from="/turtlebot2/cmd_vel" to="/hummingbird2/autopilot/velocity_command"/>
        <remap from="/turtlebot3/cmd_vel" to="/hummingbird3/autopilot/velocity_command"/>
        <remap from="/turtlebot4/cmd_vel" to="/hummingbird4/autopilot/velocity_command"/>
        <remap from="/turtlebot5/cmd_vel" to="/hummingbird5/autopilot/velocity_command"/>
        <remap from="/turtlebot6/cmd_vel" to="/hummingbird6/autopilot/velocity_command"/>
        <remap from="/turtlebot7/cmd_vel" to="/hummingbird7/autopilot/velocity_command"/>
        <remap from="/turtlebot8/cmd_vel" to="/hummingbird8/autopilot/velocity_command"/>
        <remap from="/turtlebot9/cmd_vel" to="/hummingbird9/autopilot/velocity_command"/>
        <remap from="/turtlebot10/cmd_vel" to="/hummingbird10/autopilot/velocity_command"/>
        <remap from="/turtlebot11/cmd_vel" to="/hummingbird11/autopilot/velocity_command"/>
        <remap from="/turtlebot12/cmd_vel" to="/hummingbird12/autopilot/velocity_command"/>
        <remap from="/turtlebot13/cmd_vel" to="/hummingbird13/autopilot/velocity_command"/>
        <remap from="/turtlebot14/cmd_vel" to="/hummingbird14/autopilot/velocity_command"/>
        <remap from="/turtlebot15/cmd_vel" to="/hummingbird15/autopilot/velocity_command"/>
        <remap from="/turtlebot16/cmd_vel" to="/hummingbird16/autopilot/velocity_command"/>
        <remap from="/turtlebot17/cmd_vel" to="/hummingbird17/autopilot/velocity_command"/>
        <remap from="/turtlebot18/cmd_vel" to="/hummingbird18/autopilot/velocity_command"/>
        <remap from="/turtlebot19/cmd_vel" to="/hummingbird19/autopilot/velocity_command"/>
        <remap from="/turtlebot20/cmd_vel" to="/hummingbird20/autopilot/velocity_command"/>
    </node>

    
</launch>
